microsoft/vscode
# Add persistent visual labels to find input

## About this change

This change implements persistent visual labels in accordance with [WCAG 2.1 success criteria 3.3.2](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html), notably addresses the issue that placeholder text does not provide sufficient labels for input fields as described here https://github.com/w3c/wcag/issues/864 and in other sources. This PR serves as an example of a potential change which could address the issue. It may need further review or discussion with UX. Otherwise, if the change sufficiently addresses the issue, it may be merged.

Similar issues have even been addressed in other Microsoft projects, ex.
https://github.com/microsoft/pxt-microbit/issues/2332

## Issue:

#167870

## How to test:

1. cmd-f / ctrl-f to open find widget within the editor pane
1. shift tab to replace expander and activate to show replace field
1. use screen readers on supported platforms to verify labels are properly announced

## Screenshots:

Electron

![Screenshot from 2023-01-18 15-32-29](https://user-images.githubusercontent.com/13710580/213331428-95107fb9-2ced-44da-8c86-53b6f012a918.png)

Web

![Screen Shot 2023-01-18 at 4 34 16 PM](https://user-images.githubusercontent.com/13710580/213331875-fdb91a14-c81d-40e6-b9bb-e0228b966379.png)
